api/handlers.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/ONSdigital/dp-sessions-api/session"
+	"github.com/ONSdigital/log.go/log"
+	"net/http"
+)
+
+// CreateSessionHandlerFunc returns a function that generates a session. Method = "POST"
+func CreateSessionHandlerFunc() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		var sess session.Session
+		err := json.NewDecoder(r.Body).Decode(&sess)
+		if err != nil {
+			log.Event(ctx, "unable to decode session", log.Error(err), log.ERROR)
+			http.Error(w, "Failed to decode session json", http.StatusBadRequest)
+			return
+		}
+
+		log.Event(ctx, "session created", log.Data{"session": sess}, log.INFO)
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
```

Comment:
This endpoint needs to return the session that was created/allow the caller to retrieve it.

I can see this being contentious within the team but adhering strictly to REST principals this should return a `Content-Location` header in the response with a URI to the created resource.

```
Content-Location: /session/{ID}
```
The caller can use this uri/ID to make a `GET: /session{ID)` request to get the session entity back if they require it.

However, most of our APIs in this case would just return the session body from the `POST` request - personally I prefer the first approach. This way feels like we are bastardising a well defined standard.

The caller does have to potentially make multiple requests but endpoints are well defined and **perform a single action**. Their responsibility is clearly defined and tests are easier to write. Additionally I think you end up with a much more flexible API as the actions are specific enough it enables the caller can chain them together in the way they desire to meet their need.